src/main/java/it/gov/pagopa/gpd/payments/pull/repository/PaymentPositionRepository.java

```diff
@@ -8,17 +8,34 @@
 import java.time.LocalDate;
 import java.util.List;
 
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
 @ApplicationScoped
 public class PaymentPositionRepository implements PanacheRepository<PaymentPosition> {
 
-    public static final String GET_VALID_POSITIONS_BY_TAXCODE_BASE =
-            "from PaymentPosition AS ppos Where ppos.fiscalCode = ?1 " +
-                    "AND ppos.status IN ('VALID', 'PARTIALLY_PAID') AND ppos.pull = true";
-    public static final String GET_VALID_POSITIONS_BY_TAXCODE_AND_DUE_DATE =
+    @ConfigProperty(name = "app.payment_pull.keep_aca", defaultValue = "true")
+    public boolean keepAca;
+
+    // the pull flag has no business value -> for future needs
+    private static final String BASE_QUERY = 
             "from PaymentPosition AS ppos Where ppos.fiscalCode = ?1 " +
-                    "AND ppos.status IN ('VALID', 'PARTIALLY_PAID') AND ppos.pull = true " +
-                    "AND EXISTS (from ppos.paymentOption AS po WHERE po.dueDate >= ?2)";
+            "AND ppos.status IN ('VALID', 'PARTIALLY_PAID') AND ppos.pull = true";
+
+    private static final String DUE_DATE_QUERY =
+            BASE_QUERY + " AND EXISTS (from ppos.paymentOption AS po WHERE po.dueDate >= ?2)";
 
+    public String buildQueryBase() {
+        return keepAca
+                ? BASE_QUERY + " AND ppos.serviceType IN ('ACA', 'GPD')"
+                : BASE_QUERY + " AND ppos.serviceType = 'GPD'";
+    }
+
+    public String buildQueryWithDueDate() {
+        return keepAca
+                ? DUE_DATE_QUERY + " AND ppos.serviceType IN ('ACA', 'GPD')"
+                : DUE_DATE_QUERY + " AND ppos.serviceType = 'GPD'";
+    }
+    
```

Comment:
why not join these methods?

```suggestion
    public String buildQuery(String query) {
        return keepAca
                ? query + " AND ppos.serviceType IN ('ACA', 'GPD')"
                : query + " AND ppos.serviceType = 'GPD'";
    }
    
```

Reply:
Yes I can improve as you suggested. Thanks